### ISSUE TITLE
Feature Requests: check statistics 

### ISSUE THREAD

**Issue by jeanpaulrh** [healthchecks/healthchecks#773]
Hi, maybe I did not found what’s already there but I would like to have a percentage statistic of my check. 
I can see the logs but for a check that runs quite often it’s difficult to find KO status or even have a glance of the success rate of the check maybe filtered for a given time period (1d, 1w, 1mo, 1y)
do you think it’s useful and possible?

**Comment by cuu508** [comment]
Hi @jeanpaulrh,

In the check's details page, "Current Status" section, you can see the uptime statistics for the current month and the previous two months:

![image](https://user-images.githubusercontent.com/661859/212033495-87e6fffe-f1c1-406f-882b-6b64bfb2e681.png)

Does that help?

**Comment by jeanpaulrh** [comment]
Yes. Thank you. I missed it!

**Comment by SofianeHamlaoui** [comment]
@cuu508 you think it'll be cool to have a percentage of failed & succeeded tasks ?

**Comment by cuu508** [comment]
@SofianeHamlaoui not sure, can you describe your idea a little more – how would be the percentage calculated, and where would it be displayed?

**Comment by SofianeHamlaoui** [comment]
> @SofianeHamlaoui not sure, can you describe your idea a little more – how would be the percentage calculated, and where would it be displayed?

Example : in **Dec 2022** it was down **3 times** and the period is **1 week**, means it failed 3 week checks, which means 3/4 of the checks failed.
Result : 75% failed, 25% succeeded.

Just a suggestion of course.

**Comment by cuu508** [comment]
Let's look at a few more examples.

Example I: at the start of Dec 2022, a check is **up**. During Dec 2022, the check receives **no pings**. So it changes state to **down** at the end of the first week, and stays down. What would be the success/failure percentages?

Example II: clients can explicitly signal failure. During entire Dec 2022, the check receives a failure signal every morning and a success signal every evening. In total, the check flips state from **up** to **down** 31 times. What would be the success/failure percentages?

Example III: at **Dec 15, 2022**, administrator changes the check's period to **1 day**. At the end of the month, when calculating success/failure percentages, do we use 1 week as the period, or 1 day?

**Comment by SofianeHamlaoui** [comment]
> Let's look at a few more examples.
> 
> Example I: at the start of Dec 2022, a check is **up**. During Dec 2022, the check receives **no pings**. So it changes state to **down** at the end of the first week, and stays down. What would be the success/failure percentages?
> 
> Example II: clients can explicitly signal failure. During entire Dec 2022, the check receives a failure signal every morning and a success signal every evening. In total, the check flips state from **up** to **down** 31 times. What would be the success/failure percentages?
> 
> Example III: at **Dec 15, 2022**, administrator changes the check's period to **1 day**. At the end of the month, when calculating success/failure percentages, do we use 1 week as the period, or 1 day?

Example I : depends what day of Dec the check is down, so you can have the 31days of Dec as 100% means in this case it'll be ( let's say it went down the 6th day of the first week ) so it'll be round(6*100/31) = **19% UP** & **81% DOWN** 

Example II : emm very good example, probably calculate the % of each day and then calculate the UP/Down per month ? but it can happen even if this is ment for cron jobs monitoring ?

Example III: joins the answer of Example 2 you are right, probably it'll be better if we calculate the **success/failure percentages** per day and than use them to have the month percentage. 

Probably needs more digging, but if you need help to do this I'm open to it.

**Comment by cuu508** [comment]
OK, sounds like we would be calculating the percentage based on the **time the check spent in down state**, not based on the number of "down" events.

This would be fairly easy to do, since we already calculate the time:

![image](https://user-images.githubusercontent.com/661859/220301242-ce489dd7-f61d-47b5-a80c-33be38148472.png)

In the above example, where it says "21 hours total", we could calculate the uptime percentage as 21 / (31 * 24) = 2.8%, and show it  as: "2 downtimes, 21 hours total (97.2% uptime)"

**Comment by SofianeHamlaoui** [comment]
> 

yes, this would be it ! **{{ down_duration_secs|hc_approx_duration }}** / ( 24 * number of days ) 

for the number of days you can always use **calendar.monthrange** 

`monthrange(2023, 2)[1]` > `28`

**Comment by cuu508** [comment]
OK, but would there be much benefit from displaying the uptime as percentage? The user can already do the same calculation –  approximately in head, or precisely using a calculator. 

I'm interested in use cases – how often you look up the uptime percentage, and what would you use it for?

(If it's just a fun statistic for the sake of curiosity, I'd rather not clutter the UI)

**Comment by cuu508** [comment]
I've added uptime percentage display. Looks like so:

![image](https://github.com/healthchecks/healthchecks/assets/661859/74550681-4770-4744-9fb6-9b2646aa9cd3)
